web/viewer.js

```diff
@@ -297,6 +297,12 @@ var PDFViewerApplication = {
         }
         PDFJS.externalLinkTarget = value;
       }),
+      Preferences.get('withCredentials').then(function resolved(value) {
+        if (PDFJS.withCredentials === true) {
+          return;
+        }
+        PDFJS.withCredentials = value;
```

Comment:
It feels like this preference has same use case as "disableHistory" and "externalLinkTarget", they are both very much UI related. It is very important to move "withCredentials" level up, for easier configuration. The devs who integrate pdf.js on the website, let say ```www.site.com``` and pdf comes form ```assets.site.com```, would definitely benefit from this. @yurydelendik maybe I am missing something, I am happy to update PR, can you please expiate more what do you mean?